x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/validators/debounce_async.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Unlike lodash's debounce, which resolves intermediate calls with the most
+ * recent value, this implementation waits to resolve intermediate calls until
+ * the next invocation resolves.
+ *
+ * @param fn an async function
+ *
+ * @returns A debounced async function that resolves on the next invocation
+ */
+export const debounceAsync = <Args extends unknown[], Result>(
+  fn: (...args: Args) => Result,
```

Comment:
`debounceAsync` isn't a new function. I copied it from `x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/eql_query_bar/validators.ts` since it's used in the Define Rule Schema. Since it already worked I didn't spend time on refactoring rather focusing on ES|QL functionality.

Indeed the implementation and typing looks suboptimal. I finally spent some time to refactor the function and improve the typing.